src/platform/windows/mod.rs

```diff
@@ -0,0 +1,1482 @@
+            let header: [u32; 2] = [
+                if big_data_sender.is_some() { 0 } else { data.len() } as u32,
+                oob_data.len() as u32
+            ];
+            let header_bytes: &[u8] = slice::from_raw_parts(header.as_ptr() as *const u8, HEADER_SIZE);
+
+            // We need to write the main message first, which will indicate to the receiver
+            // that it should receive a message on the OOB big data channel.  If we don't
+            // send this first we'll block
+            try!(write_buf(*self.handle, &header_bytes));
+            if big_data_sender.is_none() {
+                try!(write_buf(*self.handle, data));
+            }
+            try!(write_buf(*self.handle, &oob_data));
+            if big_data_sender.is_some() {
+                dd2!("[c {:?}] sending unfragmented to big-data channel: {} bytes",
+                     *self.handle, data.len());
+                try!(big_data_sender.unwrap().send_unfragmented(data));
+            }
+        }
+
+        Ok(())
+    }
+}
+
+pub struct OsIpcReceiverSet {
+    // the IOCP that we select on
+    iocp: WinHandle,
+
+    // the set of receivers
+    receivers: Vec<OsIpcReceiver>,
+}
+
+pub enum OsIpcSelectionResult {
+    DataReceived(i64, Vec<u8>, Vec<OsOpaqueIpcChannel>, Vec<OsIpcSharedMemory>),
+    ChannelClosed(i64),
+}
+
+impl OsIpcReceiverSet {
+    pub fn new() -> Result<OsIpcReceiverSet,WinError> {
+        unsafe {
+            let iocp = kernel32::CreateIoCompletionPort(INVALID_HANDLE_VALUE,
+                                                        ptr::null_mut(),
+                                                        0 as winapi::ULONG_PTR,
+                                                        0);
+            if iocp.is_null() {
+                return Err(WinError::last("CreateIoCompletionPort"));
+            }
+
+            Ok(OsIpcReceiverSet {
+                iocp: WinHandle::new(iocp),
+                receivers: vec![],
+            })
+        }
+    }
+
+    pub fn add(&mut self, mut receiver: OsIpcReceiver) -> Result<i64,WinError> {
+        // use this to identify the receiver
+        let receiver_handle = *receiver.handle;
+
+        try!(receiver.add_to_iocp(*self.iocp));
+
+        // FIXME we *may* need a mutex to protect self.receivers --
+        // one thread could be adding something to this Set while
+        // another is calling select(); the add() could cause the
+        // receivers array to reallocate while we're doing stuff with
+        // it in select().  That would mean an add() would block while
+        // a select() is blocking.
+        // 
+        // A better option would be to have a mutex around a
+        // self.receivers_to_add array, and have select drain those
+        // and append to self.receivers whenever it's called.
+        self.receivers.push(receiver);
+
+        dd!("[# {:?}] ReceiverSet add {:?}", *self.iocp, receiver_handle);
+
+        Ok(receiver_handle as i64)
```

Comment:
I know Windows does reuse `HANDLE` values, but I'm also not sure how likely this is. The implementation here would probably be similar to what is done in the `unix` module.